openmicroscopy/openmicroscopy
# Img data json parents

As suggested by @chris-allan in https://github.com/openmicroscopy/openmicroscopy/pull/4446#issuecomment-185231212, this fixes the handling of multiple parents (Datasets & Wells) by ```marshalImage()```.

NB: I haven't changed handling of Projects yet.

@chris-allan suggested the "first" of multiple parents be used for keys such as "datasetId", although I'm not sure how we want to sort datasets before picking the first?

To test:
 - Easy: put an image into multiple Datasets then inspect the json object at url: ```webgateway/imgData/3731/```

You should see under the ```meta``` section a ```datasets``` list like this:
```
datasets: [
{
    description: "",
    id: 601,
    name: "CSFV"
},
{
    description: "Images in this Dataset are from ROIs of parent Image: Name: scram_01.r3d Image ID:   324",
    id: 3301,
    name: "From_ROIs"
}
],
```
and also ```datasetName```, ```datasetId``` and ```datasetDescription``` based on the first dataset.

 - Harder: Run the "Dataset_To_Plate" script without removing the image from the Dataset.
Now check that you see multiple ```wells``` in the ```meta``` as above.
